### PR TITLE
New Parser: Kiuwan SCA Scan

### PR DESCRIPTION
**Pull request by mwager**
As discussed in slack #defectdojo-dev:

grafik

**Checklist**

This checklist is for your information.

- [x] Make sure to rebase your PR against the very latest `dev`.
- [x] Features/Changes should be submitted against the `dev`.
- [x] Give a meaningful name to your PR, as it may end up being used in the release notes.
- [x] Your code is flake8 compliant.
- [x] Your code is python 3.11 compliant.
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the docs at https://github.com/DefectDojo/django-DefectDojo/tree/dev/docs as part of this PR.
- [x] Add applicable tests to the unit tests.

(cc @flmarkus)

### CODE REVIEW

File: dojo/tools/kiuwan_sca/parser.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                finding.epss_percentile = row["epss_percentile"]
+
+            if "cVSSv3BaseScore" in row:
+                finding.cvssv3_score = float(row["cVSSv3BaseScore"])
+
+            finding.references = "See Kiuwan Web UI"
+            finding.mitigation = "See Kiuwan Web UI"
+            finding.static_finding = True
+
+            key = hashlib.sha256(
```
Comment (manuel-sommer) [inline dojo/tools/kiuwan_sca/parser.py:72]: Hi @mwager, the PR looks quite good.

I don't know if you need the key at all if you choose DEDUPE_ALGO_HASH_CODE.

Comment (mwager) [reply dojo/tools/kiuwan_sca/parser.py:72]: Thx! As I understood, this is only for deduping per import? i.e. removing dupes in *one* import file, therefore preventing further deduplication later? Maybe I got this wrong then?

grafik

But I can remove it if this is not needed. We tested it with lots of prod data and it does what we expect :)

Comment (manuel-sommer) [reply dojo/tools/kiuwan_sca/parser.py:72]: Let's wait for the reviews of the maintainers. Ah, good point, maybe you are right, but why would you have duplicates in one file?

**Comment by mwager** [comment]
Any more work needed from my side? I don't understand the error from the "Detect Merge conflicts" action:

grafik

See here: https://github.com/DefectDojo/django-DefectDojo/actions/runs/8889852139/job/24429470629?pr=10064

**Comment by mtesauro** [comment]
@mwager 

> Any more work needed from my side? I don't understand the error from the "Detect Merge conflicts" action:

That "Detect Merge conflicts" is just GH having issues running GH Actions and nothing you need to do. You're good for now as there's an issue with the REST tests we're trying to figure out so hold tight for a bit and we can start the approvals once we sort out whatever is going on with the REST tests.

**Comment by mwager** [comment]
@mtesauro We really need this in master because via Kubernetes we would need to build and host our own images to use it. Without this Parser we cannot import our supply chain scans which is quite an issue. I think the Rest test Issues are fixed now?

**Comment by mtesauro** [comment]
@mwager 

This PR is targeting the dev branch so, once it's got the needed approvals, it will be merged into main/master on the first Monday in August and be part of version 2.37.0

I just approved the tests to run, assuming those are good, we can start getting the needed approvals.  Thanks for you patience - summer is 'interesting' with people going on holiday and such.

**Comment by mtesauro** [comment]
@mwager 
Hate to suggest this but it may make sense to close this PR and open a new one with the same changes. It's likely easier that trying to get this one green on the GH Action tests. We recently did several updates to brittle/flaky tests and I suspect this PR has those old tests associated with it since I just re-kicked off the tests and the same ones failed again.

**Comment by mwager** [comment]
Re-opened [here](https://github.com/DefectDojo/django-DefectDojo/pull/10522), hope this is good to go now :)

**Pull request closed without merging**